src/diffusers/loaders/unet.py

```diff
@@ -336,8 +340,108 @@ def _process_lora(self, state_dict, unet_identifier_key, network_alphas, adapter
             # otherwise loading LoRA weights will lead to an error
             is_model_cpu_offload, is_sequential_cpu_offload = self._optionally_disable_offloading(_pipeline)
 
-            inject_adapter_in_model(lora_config, self, adapter_name=adapter_name)
-            incompatible_keys = set_peft_model_state_dict(self, state_dict, adapter_name)
+
+            def _check_hotswap_configs_compatible(config0, config1):
+                # To hot-swap two adapters, their configs must be compatible. Otherwise, the results could be false. E.g. if they
+                # use different alpha values, after hot-swapping, the alphas from the first adapter would still be used with the
+                # weights from the 2nd adapter, which would result in incorrect behavior. There is probably a way to swap these
+                # values as well, but that's not implemented yet, and it would trigger a re-compilation if the model is compiled.
+
+                # TODO: This is a very rough check at the moment and there are probably better ways than to error out
+                config_keys_to_check = ["lora_alpha", "use_rslora", "lora_dropout", "alpha_pattern", "use_dora"]
```

Comment:
Should we fetch this by inspecting the `LoraConfig` class init, instead? If the `target_modules` differ that would make hotswap incompatible as well no?

Reply:
Hmm, I don't think we should compare `target_modules`. First of all, `target_modules` can be a list of str but also a regex. We can't really know if the two amount to the same result or not. Second, I think it's more robust to check the actually adapted modules.

Reply:
Makes sense. Could you help me understand how / where we're accomplishing:

> Second, I think it's more robust to check the actually adapted modules.

Reply:
In `_hotswap_adapter_from_state_dict`, we check for missing and unexpected keys. If the we're not matching the target modules, we would raise an error there.